### ISSUE TITLE
linter: false positive with String.raw for unicorn plugin

### ISSUE THREAD

**Issue by Mister-Hope** [oxc-project/oxc#20486]
### What version of Oxlint are you using?

latest

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
// Replace this with your actual config file contents.
// It can be helpful to keep it minimal, if possible, but please ensure that it actually reproduces the issue you're reporting.
{
  "categories": {
    "pedantic": "error"
  },
  "plugins": ["unicorn"],
  "rules": {
    "unicorn/escape-case": "error",
    "unicorn/no-hex-escape": "error",
  }
}
```

### What happened?

```
const content = String.raw`\x0d`;
```

It checked contents inside `String.raw`

The string should be kept as is, but one rule suggests changing it to `` String.raw`\x0D` `` and the other rule's auto fix changes to `` String.raw`\u000d` ``.

This is breaking the string content.

Playground link [here](https://playground.oxc.rs/?t=ast&lintRules=unicorn%2Fno-hex-escape%2Cunicorn%2Fescape-case&options=%7B%22run%22%3A%7B%22lint%22%3Atrue%2C%22formatter%22%3Afalse%2C%22transform%22%3Afalse%2C%22isolatedDeclarations%22%3Afalse%2C%22whitespace%22%3Afalse%2C%22mangle%22%3Afalse%2C%22compress%22%3Afalse%2C%22scope%22%3Atrue%2C%22symbol%22%3Atrue%2C%22cfg%22%3Afalse%7D%2C%22parser%22%3A%7B%22extension%22%3A%22ts%22%2C%22allowReturnOutsideFunction%22%3Atrue%2C%22preserveParens%22%3Atrue%2C%22allowV8Intrinsics%22%3Atrue%2C%22semanticErrors%22%3Atrue%7D%2C%22linter%22%3A%7B%22config%22%3A%22%7B%5C%22categories%5C%22%3A%7B%5C%22correctness%5C%22%3A%5C%22off%5C%22%7D%2C%5C%22rules%5C%22%3A%7B%5C%22unicorn%2Fno-hex-escape%5C%22%3A%5C%22error%5C%22%2C%5C%22unicorn%2Fescape-case%5C%22%3A%5C%22error%5C%22%7D%7D%22%7D%2C%22formatter%22%3A%7B%22useTabs%22%3Afalse%2C%22tabWidth%22%3A2%2C%22endOfLine%22%3A%22lf%22%2C%22printWidth%22%3A80%2C%22singleQuote%22%3Afalse%2C%22jsxSingleQuote%22%3Afalse%2C%22quoteProps%22%3A%22as-needed%22%2C%22trailingComma%22%3A%22all%22%2C%22semi%22%3Atrue%2C%22arrowParens%22%3A%22always%22%2C%22bracketSpacing%22%3Atrue%2C%22bracketSameLine%22%3Afalse%2C%22objectWrap%22%3A%22preserve%22%2C%22singleAttributePerLine%22%3Afalse%7D%2C%22transformer%22%3A%7B%22target%22%3A%22es2015%22%2C%22useDefineForClassFields%22%3Atrue%2C%22experimentalDecorators%22%3Atrue%2C%22emitDecoratorMetadata%22%3Atrue%7D%2C%22isolatedDeclarations%22%3A%7B%22stripInternal%22%3Afalse%7D%2C%22codegen%22%3A%7B%22normal%22%3Afalse%2C%22jsdoc%22%3Afalse%2C%22annotation%22%3Afalse%2C%22legal%22%3Afalse%7D%2C%22compress%22%3A%7B%7D%2C%22mangle%22%3A%7B%22topLevel%22%3Atrue%2C%22keepNames%22%3Afalse%7D%2C%22controlFlow%22%3A%7B%22verbose%22%3Afalse%7D%2C%22inject%22%3A%7B%22inject%22%3A%7B%7D%7D%2C%22define%22%3A%7B%22define%22%3A%7B%7D%7D%7D&code=const+content+%3D+String.raw%60%5Cx0d%60%3B%0A)

**Comment by camc314** [comment]
#20496 
#20497
